tests/test-webapps/test-websocket-client-provided-webapp/src/main/java/org/eclipse/jetty/tests/webapp/websocket/WebSocketClientServlet.java

```diff
@@ -0,0 +1,129 @@
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.tests.webapp.websocket;
+
+import java.io.PrintWriter;
+import java.net.URI;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.api.util.WSURI;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+
+@WebServlet("/")
+public class WebSocketClientServlet extends HttpServlet
+{
+    private WebSocketClient client;
+
+    @Override
+    public void init() throws ServletException
+    {
+        // We must rely on jetty-websocket-httpclient.xml as we do not have access to server classes like HttpClient.
+        client = new WebSocketClient();
+
+        try
+        {
+            client.start();
+        }
+        catch (Exception e)
+        {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    public void destroy()
+    {
+        try
+        {
+            client.stop();
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+    {
+        try
+        {
+            resp.setContentType("text/html");
+
+            // Send and receive a websocket echo on the same server.
+            ClientSocket clientSocket = new ClientSocket();
+            URI wsUri = WSURI.toWebsocket(req.getRequestURL()).resolve("echo");
+            client.connect(clientSocket, wsUri).get(5, TimeUnit.SECONDS);
+            clientSocket.session.getRemote().sendString("test message");
+            String response = clientSocket.textMessages.poll(5, TimeUnit.SECONDS);
+            clientSocket.session.close();
+            clientSocket.closeLatch.await(5, TimeUnit.SECONDS);
+
+            PrintWriter writer = resp.getWriter();
+            writer.println("WebSocketEcho: " + ("test message".equals(response) ? "success" : "failure"));
+            writer.println("WebSocketEcho: success");
+            // We cannot test the HttpClient timeout because it is a server class not exposed to the webapp.
+            // writer.println("ConnectTimeout: " + client.getHttpClient().getConnectTimeout());
```

Comment:
I think you can do this via reflection: 

```java
Object httpClient = client.getHttpClient();
// etc.
```

If you cannot, then remove the `connectTimeout` setting from `jetty-websocket-httpclient.xml`.